pkg/services/apiserver/builder/auth_handler.go

```diff
@@ -0,0 +1,42 @@
+package builder
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	"k8s.io/klog/v2"
+)
+
+var _ http.Handler = &authHandler{}
+
+type authHandler struct {
+	delegateHandler http.Handler
+}
+
+func newAuthHandler(delegateHandler http.Handler) http.Handler {
+	return &authHandler{
+		delegateHandler: delegateHandler,
+	}
+}
+
+func (h *authHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	signedInUser, err := appcontext.User(ctx)
+	if err != nil {
+		klog.Error("failed to get signed in user", "err", err)
+		h.delegateHandler.ServeHTTP(w, req)
+		return
+	}
+
+	req.Header.Set("X-Remote-User", strconv.FormatInt(signedInUser.UserID, 10))
```

Comment:
Do we want to make these header names configurable?

Reply:
Same question for me - it's fine to have them like this as a short-term solution, but AFAIK normally these are configurable

Reply:
afaik there is a mechanism for specifying these, can we pull them from config?
https://kubernetes.io/docs/reference/access-authn-authz/authentication/#authenticating-proxy

Reply:
i need to change the approach here because these are being overridden by an authenticator. i think the new approach should respect the config